File: src/test/java/io/plagov/adven/aoc2016/Day04Test.java
package io.plagov.adven.aoc2016;

import io.plagov.advent.aoc2016.Day04;
import org.junit.jupiter.api.Test;
import util.FileUtil;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day04Test {

    private final List<String> input = FileUtil.readInputFile("2016/day04.txt");

    @Test
    void solveFirst() {
        var day04 = new Day04();
        assertEquals(173787, day04.partOne(input));
    }
}

File: src/test/kotlin/io/plagov/advent/aoc2024/Day04Test.kt
package io.plagov.advent.aoc2024

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

import util.readInputFileAsString

class Day04Test {

  private val input = readInputFileAsString("2024/day04.txt")

  private val day4 = Day04()

  @Test
  fun partOne() { 
    Assertions.assertEquals(2557, day4.partOne(input))
  }

  @Test
  fun partTwo() {
    Assertions.assertEquals(1854, day4.partTwo(input))
  }
}
